doc/plutus/architecture/constraints/index.rst

```diff
@@ -0,0 +1,294 @@
+However, when using this constraint to generate a transaction, it will:
+
+1- sum up the `Value` of the transaction outputs.
+2- add the missing `Value` through a new transaction output if the sum is less than the provided `Value`
+
+Then, it will need to know who pays (`PaymentPubKeyHash`) for that transaction output.
+Thus, this `PaymentPubKeyHash` needs to be provided through some other mean.
+In our case, it will be provided through a lookups value.
+
+Limitations
+^^^^^^^^^^^
+
+There is one major drawback to this approach: the lookups value in transaction creation step.
+To create a transaction, you need to provide the required lookups that are needed by the constraints.
+The problem is that there is nothing linking a specific constraint to a specific lookup.
+More spefically, there is nothing in the API (apart from reading the documentation) which prevents the users from providing an unrequired lookup, or forgetting to provide a required lookup.
+The result is a runtime error, and not a compile error.
+
+To illustrate the point, there is nothing preventing the user from doing the following:
+
+.. code-block:: haskell
+
+  mkMyTx1 :: PaymentPubKeyHash -> TxOutRef -> Either MkTxError UnbalancedTx
+  mkMyTx1 pkh ref =
+    let lookups = mempty -- Missing lookup for the transaction output reference
+     in mkTx lookups (mustSpendPubKeyOutput ref)
+
+  mkMyTx2 :: PaymentPubKeyHash -> TxOutRef -> Either MkTxError UnbalancedTx
+  mkMyTx2 pkh ref =
+    let lookups = ownPaymentPubKeyHash pkh -- Unnecessary lookup
+     in mkTx lookups (mustSpendPubKeyOutput ref)
+
+Both of these functions will return a `Left` value.
+
+Alternatives
+^^^^^^^^^^^^
+
+Integrating lookups in the constraints
+""""""""""""""""""""""""""""""""""""""
+
+The data constructors of `TxConstraint` could contain all the needed information for both the Plutus script and the transaction creation.
+
+For example, we could replace `TxConstraint` with:
+
+.. code-block:: haskell
+
+  data TxConstraint =
+      MustBeSignedBy PaymentPubKeyHash (Maybe PaymentPubKey)
+      MustSpentPubKeyOutput TxOutRef ChainIndexTxOut
+      MustSpendAtLeast PaymentPubKeyHash Value
+      -- ... (OTHER CONSTRAINTS)
+
+On the upside:
+
+* the lookups are integrated in the constraints itself for the transaction creation.
+
+On the downside:
+
+* some of these additionnal types cannot be used in Plutus scripts such as `ChainIndexTxOut`.
+* some of these additionnal types might increase cost of Plutus scripts. TODO: to validate
+* the users has the burden of handling runtime errors such as not finding the `ChainIndexTxOut` of a `TxOutRef` from an external source.
+
+Distinct isomorphic constraints
+"""""""""""""""""""""""""""""""
+
+Instead of a single `TxConstraint` which is used for both Plutus scripts and creating transactions, we can create two distinct and isomorphic datatypes. TODO: Not sure if isomorphic is the right word here.
```

Comment:
I'd just say "different" or "parallel" if you want to highlight their connection. "isomorphic" is a technical term and we are not really using it in its technical meaning.